self-hosting/prepare-system/ubuntu/prepare.sh

```diff
@@ -21,7 +21,12 @@ sudo useradd -d /home/medic -m medic
 sudo mkdir -p /home/medic/self-hosting/main
 
 sudo cp ../../main/docker-compose.yml /home/medic/self-hosting/main/
-echo "HA_PASSWORD=$(sudo cat /srv/storage/medic-core/passwd/admin)" >  /home/medic/self-hosting/main/.env
+
+HA_FILE=/srv/storage/medic-core/passwd/admin
```

Comment:
I’m forgetting where in the fresh install process this admin file gets written, but it should have cascaded down from the variable you set. There’s a deeper medic-os issue going on. Although, if this is working, it should keep working for all our other scenarios, so it’s not the biggest worry.

Reply:
@Hareet  Now that I've been digging into docker-compose a bit more. I should not have made this change.  `docker-compose` takes the `.env` file from the location where the `docker-compose` command is executed. In our case supervisor is executing from a different directory. This means the .env file is ignored completely when it exists at `/home/medic/self-hosting/main/.env` and needs to be placed at `/home/medic/self-hosting/` where supervisor is executing `docker-compose` from for it to be recognized. I kinda feel like the supervisor.conf directory needs to be changed but I am not sure what that would impact.

Reply:
@newtewt Yup, you are right. You are free to do anything with the .env file since variables defined in the compose file will take precedence. My process won't be effected at all. 

I don't touch the .env file and define those explicitly inside the compose.yml that supervisor launches and during the capacity building sessions I walk through all the declarations in that file.

You could either use sed to in-line changes to the compose file before launching or point the .env file to the same launch directory defined in supervisor.conf